### PR TITLE
Enhancement/8978 Refactor UnsatisfiedScopesAlert

### PR DESCRIPTION
**Pull request by jimmymadon**
## Summary

Addresses issue:

- #8978 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by tofumatt — CHANGES_REQUESTED** [review]
I just have a few comments on the code itself, all quite minor.

But when I tried to get the Gathering data notification to appear by setting up Site Kit on my development site with the Tester Plugin, I never saw the notification 🤔 

Based on the QA Brief I expected it to appear, but nothing did.

File: assets/js/components/notifications/ErrorNotifications.js [hunk 1]
```diff
@@ -133,14 +137,15 @@ export default function ErrorNotifications() {
...
-					{ ! showUnsatisfiedScopesAlertGTE && (
-						<UnsatisfiedScopesAlert />
-					) }
 					{ showUnsatisfiedScopesAlertGTE && (
 						<UnsatisfiedScopesAlertGTE />
 					) }
 				</Fragment>
 			) }
+			<Notifications
+				viewContext={ viewContext }
```
[the remark this answers is not in the document]
Comment (jimmymadon) [reply assets/js/components/notifications/ErrorNotifications.js:146]: This has been lingering on from my first issue - I wanted to refactor this and kept postponing it! But just did this now! Thanks!

File: assets/js/components/notifications/UnsatisfiedScopesAlert.js [hunk 2]
```diff
@@ -195,18 +197,21 @@ export default function UnsatisfiedScopesAlert() {
...
-			ctaLink={ connectURL }
-			onCTAClick={ () => {
-				doingCTARef.current = true;
-			} }
-			ctaLabel={ ctaLabel }
-		/>
+		<Notification>
+			<NotificationError
+				title={ title }
+				description={ <Description description={ message } /> }
```
[the remark this answers is not in the document]
Comment (jimmymadon) [reply assets/js/components/notifications/UnsatisfiedScopesAlert.js:203]: Haha - renamed to text!

File: assets/js/googlesitekit/notifications/components/common/ActionsCTALinkDismiss.js [hunk 3]
```diff
@@ -36,39 +34,18 @@ export default function ActionsCTALinkDismiss( {
...
-				disabled={ isNavigatingToCTALink }
-			>
-				{ ctaLabel }
-			</SpinnerButton>
+			<CTALink
+				id={ id }
+				ctaLink={ ctaLink }
+				ctaLabel={ ctaLabel }
+				dismissExpires={ dismissExpires }
+			></CTALink>
```
[the remark this answers is not in the document]
Comment (jimmymadon) [reply assets/js/googlesitekit/notifications/components/common/ActionsCTALinkDismiss.js:48]: Thanks!

**Comment by jimmymadon** [comment]
@tofumatt Thanks for the review. This seems better now. Was going crazy with all the testing so missed checking the remaining notifications after my last few modifications.

**Review by tofumatt — APPROVED** [review]
Looks good 👍🏻

**Comment by tofumatt** [comment]
Screenshot 2024-09-02 at 16 40 13

VRT failure here is totally unrelated, so I'm fixing it here then merging this. 😅

**Pull request merged**